unity/EditorPlugin/NonUnity/RiderProtocolController.cs

```diff
@@ -46,7 +48,15 @@ public ProtocolInstance(int port, string solutionName)
 
     public static string ToJson(List<ProtocolInstance> connections)
     {
-        return JsonConvert.SerializeObject(connections);
+        //return JsonConvert.SerializeObject(connections); //turns out to be slow https://github.com/JetBrains/resharper-unity/issues/728 
+      var sb = new StringBuilder("[");
+
+      sb.Append(connections
+        .Select(connection=> "{" + $"\"Port\":{connection.Port},\"SolutionName\":\"{connection.SolutionName}\"" + "}")
+        .Aggregate((a, b) => a + "," + b));
+
+      sb.Append("]");
+      return sb.ToString();
     }
```

Comment:
A `foreach` might be even better here - LINQ can be quite expensive.

Reply:
I don't like removing trailing comma, Aggregate is safer. I can hardly imagine someone having more than 3-4 solution files in this folder.